src/Bicep.RegistryModuleTool.UnitTests/Extensions/IFileExtensionsTests.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.RegistryModuleTool.Extensions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.IO.Abstractions.TestingHelpers;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.RegistryModuleTool.UnitTests.Extensions
+{
+    [TestClass]
+    public class IFileExtensionsTests
+    {
+        [TestMethod]
+        public void CreateTempFile_Disposed_DeletesTempFile()
+        {
+            // Arrange.
+            var fileSystem = new MockFileSystem();
+
+            // Act.
+            using (var tempFile = fileSystem.File.CreateTempFile())
+            {
+                using (fileSystem.File.Create(tempFile.Path)) { }
+            }
+
+            // Assert.
+            fileSystem.AllFiles.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void CreateTempFile_GarbageCollected_DeletesTempFile()
+        {
+            // Arrange.
+            var fileSystem = new MockFileSystem();
+
+            var action = new Action(() =>
+            {
+                var tempFile = fileSystem.File.CreateTempFile();
+
+                using (fileSystem.File.Create(tempFile.Path)) { }
+            });
+
+            // Act.
+            action();
+            GC.Collect();
```

Comment:
>Collect [](http://example.com/codeflow?start=15&length=7)

It's not 100% guaranteed that everything that is collectable will be collected when you call it. Is there a way to the need for this?

Reply:
Nope, unless this is some kind of common library that many will use. Removed the test and the extension class.